corpus/transform.py

```diff
@@ -337,11 +339,102 @@ def run(self):
+        add_sc = {sc.name: sc for sc in to_add.top_level_subcorpora()}
+        for name, sc in add_sc.items():
+            if name in base_sc:
+                MergeCorporaJob.merge_corpora(base_sc[name], sc)
+            else:
+                base.add_subcorpus(sc)
+
+        return base
+
+    @staticmethod
+    def merge_recordings(base: corpus.Recording, to_add: corpus.Recording):
+        """
+        Merges a given base recording with another recording.
+
+        :param base: Base recording.
+        :param to_add: Recording to be merged into :param:`base`.
+        """
+        assert base.audio == to_add.audio, "Same recording name points do different audio files"
+
+        # Merge the speakers.
+        base_spk = {spk.name: spk for spk in base.speakers}
+        to_add_spk = {spk.name: spk for spk in to_add.speakers}
+        for name, spk in to_add_spk.items():
+            if name in base_spk:
+                assert spk.__dict__ == base_spk[name].__dict__, (
+                    f"Found same speaker {name} with different attributes in the base/to be added corpus.\n"
+                    f"Speaker attributes (base corpus): {base_spk[name]}.\n"
+                    f"Speaker attributes (corpus to be added): {spk}."
+                )
+            else:
+                base.speakers[name] = spk
+
+        # Merge the segments.
+        base_seg = {seg.name: seg for seg in base.segments}
+        add_seg = {seg.name: seg for seg in to_add.segments}
+        for name, seg in add_seg.items():
+            if name in base_seg:
+                # Only compare segments with respect to their actual segment data.
+                # Don't include the recording to which they belong, since it's bound to change.
+                # Not only because it's a dynamic object, but also because it comes from a completely different corpus.
+                # In practice, the equality check would yield False when comparing segments with recording objects.
+                seg_without_rec = {key: value for key, value in seg.__dict__.items() if key != "recording"}
+                base_seg_without_rec = {
+                    key: value for key, value in base_seg[name].__dict__.items() if key != "recording"
+                }
+                assert seg_without_rec == base_seg_without_rec, (
+                    "The same segment exists in two corpora with different parameters. "
+                    "Please filter either of the two segments in the corpora to be merged. "
+                    f"Affected segment name: {name}.\n"
+                    f"Affected recording (base): {base_seg[name].recording.fullname()}.\n"
+                    f"Affected recording (to be added): {seg.recording.fullname()}.\n"
+                    f"Affected corpus (base): {base_seg[name].recording.corpus.fullname()}.\n"
+                    f"Affected corpus (to be added): {seg.recording.corpus.fullname()}.\n"
```

Comment:
I thought I had commented on that before, but I can't find it any more, so sorry if duplicate:

The lines with affected corpus are redundant as `recording.fullname()` will just return `recording.corpus.fullname() + '/' + recording.name` so all information is already contained in the recording print.
And one could even just print `seg.fullname()` and `base_seg[name].fullname()`

Besides, by construction of the merge algorithm, both fullnames should coincide.